base10.go

```diff
@@ -0,0 +1,129 @@
+package uint256
+
+import (
+	"strconv"
+	"strings"
+)
+
+const twoPow256Sub1 = "115792089237316195423570985008687907853269984665640564039457584007913129639935"
+const twoPow128 = "340282366920938463463374607431768211456"
+const twoPow64 = "18446744073709551616"
+
+func (z *Int) Base10() string {
+	return z.ToBig().String()
+}
+
+// SetString implements a subset of (*big.Int).SetString
+// ok will be true iff i == nil
+func (z *Int) SetString(s string, base int) (i *Int, ok bool) {
+	switch base {
+	case 0:
+		if strings.HasPrefix(s, "0x") {
+			err := z.fromHex(s)
+			if err != nil {
+				return nil, false
+			}
+			return z, true
+		}
+		err := z.SetFromBase10(s)
+		if err != nil {
+			return nil, false
+		}
+		return z, true
+	case 10:
+		err := z.SetFromBase10(s)
+		if err != nil {
+			return nil, false
+		}
+		return z, true
+	case 16:
+		err := z.fromHex(s)
+		if err != nil {
+			return nil, false
+		}
+		return z, true
+	}
+	return nil, false
+}
+
+// FromBase10 is a convenience-constructor to create an Int from a
+// decimal (base 10) string. Numbers larger than 256 bits are not accepted.
+func FromBase10(hex string) (*Int, error) {
+	var z Int
+	if err := z.SetFromBase10(hex); err != nil {
+		return nil, err
+	}
+	return &z, nil
+}
+
+// SetFromBase10 sets z from the given string, interpreted as a decimal number.
+func (z *Int) SetFromBase10(s string) (err error) {
+	if len(s) < len(twoPow256Sub1) {
+		return z.fromBase10Long(s)
+	}
+	if len(s) == len(twoPow256Sub1) {
+		if s > twoPow256Sub1 {
+			return ErrBig256Range
+		}
+		return z.fromBase10Long(s)
+	}
+	return ErrBig256Range
+}
+
+var scaleTable10 [78]Int
+
+func init() {
+	for k := range scaleTable10 {
+		scaleTable10[k].Exp(NewInt(10), NewInt(uint64(k)))
+	}
+}
+
+func (z *Int) fromBase10Long(bs string) error {
+	z.Clear()
+	if bs == "" {
+		return nil
+	}
+	iv := 19
+	c := 0
+	if len(bs) >= (iv * 4) {
```

Comment:
Could you add a bit of description what happens inside this method -- what algorithm you're following? 
Doesn't have to be a lot, but a brief description. 
Also, what does `iv` mean? I associate the term with cryptographic  input vector, but this is probably something else? 
Also, is the unrolled loop really worth it? If there's no tangible gain, I'd prefer to have it in a shorter form

Reply:
lol. i think iv originally meant "index value" or something like that.

I've cleaned up the function and added some comments.  - let me know if it's still unclear